partition_snapshot_row_cursor.hh

```diff
@@ -198,17 +268,41 @@ class partition_snapshot_row_cursor final {
     // When throws, the cursor is invalidated and its position is not changed.
     bool advance(bool keep) {
         memory::on_alloc_point();
-        position_in_version::less_compare heap_less(_schema);
+        version_heap_less_compare heap_less(*this);
         assert(iterators_valid());
         for (auto&& curr : _current_row) {
             if (!keep && curr.unique_owner) {
                 mutation_partition::rows_type::key_grabber kg(curr.it);
                 kg.release(current_deleter<rows_entry>());
+                if (_reversed && curr.it) [[unlikely]] {
+                    if (curr.rows->begin() == curr.it) {
+                        _background_continuity |= bool(curr.it->continuous());
+                        curr.it = {};
+                    } else {
+                        curr.continuous = curr.it->continuous();
+                        --curr.it;
+                    }
+                }
             } else {
-                ++curr.it;
+                if (_reversed) [[unlikely]] {
+                    if (curr.rows->begin() == curr.it) {
+                        _background_continuity |= bool(curr.it->continuous());
+                        curr.it = {};
+                    } else {
+                        curr.continuous = curr.it->continuous();
+                        --curr.it;
+                    }
+                } else {
+                    ++curr.it;
+                    if (curr.it) {
+                        curr.continuous = curr.it->continuous();
+                    }
```

Comment:
This was not present in the previous version and it does not seem related to reversed queries. Why does it have to be added?

Reply:
`curr.continuous` is related to reversed queries. We also discuss this in this thread: https://github.com/scylladb/scylla/pull/9454#discussion_r763213161, check it out.

Reply:
curr.continuous is a new thing. I added it so that it uniformly reflects cursor's continuity from both forward and reverse perspective. it->continuous() reflects only forward perspective (table order)